astrapy/core/db.py

```diff
@@ -54,6 +55,74 @@
 logger = logging.getLogger(__name__)
 
 
+class _PrefetchIterator(Iterator[API_DOC]):
+    def __init__(
+        self,
+        prefetched: int,
+        request_method: PaginableRequestMethod,
+        options: Optional[Dict[str, Any]],
+        raw_response_callback: Optional[Callable[[Dict[str, Any]], None]] = None,
+    ):
+        self.queue: queue.Queue[Optional[API_DOC]] = queue.Queue(prefetched)
+        self.request_method = request_method
+        self.options = options
+        self.raw_response_callback = raw_response_callback
+        self.initialised = threading.Event()
+        self.stop = threading.Event()
+        self.thread = threading.Thread(
+            target=_PrefetchIterator.queued_paginate, args=(weakref.proxy(self),)
+        )
+        self.thread.start()
+        while not self.initialised.is_set():
```

Comment:
wouldn't this loop consume many resources without a (tiny) sleep in its body or something?
I think `Event.is_set()` does not involve any thread-synch, it would continuously read and read and read (?)

Reply:
I copied that from somewhere.
But `self.initialised.wait()` would probably be much better.

Reply:
done.